### DISCUSSION TITLE
Linprog gives x values not in bounds

### DISCUSSION THREAD

**Discussion by Keshabl** [yalmip/YALMIP#1414] [Problems using YALMIP]
Im trying to make code that can solve sudoku by using this math model

![image](https://github.com/yalmip/YALMIP/assets/83855659/1c6158eb-aa76-4e89-bcc9-e10a5abc8d51)

i,j,k are raws, columns and values

so i wrote code using it

`
import numpy as np
from scipy.optimize import linprog
import networkx as nx
import matplotlib.pyplot as plt

S=1
sud1   = [[2, 0, 5, 0, 0, 9, 0, 0, 4],
          [0, 0, 0, 0, 0, 0, 3, 0, 7],
          [7, 0, 0, 8, 5, 6, 0, 1, 0],
          [4, 5, 0, 7, 0, 0, 0, 0, 0],
          [0, 0, 9, 0, 0, 0, 1, 0, 0],
          [0, 0, 0, 0, 0, 2, 0, 8, 5],
          [0, 2, 0, 4, 1, 8, 0, 0, 6],
          [6, 0, 8, 0, 0, 0, 0, 0, 0],
          [1, 0, 0, 2, 0, 0, 7, 0, 8]]

c = [0]*9*9*9*S

for i in range (0,9):
    for j in range(0, 9):
        if sud1[i][j] != 0:
            c[i*81+j*9+sud1[i][j]-1] = 1

Aeq10 = np.zeros((81,9*9*9*S))
for j in range (1,10):
    for k in range (1,10):
        for i in range (1,10):
            Aeq10[(j-1)*9+(k-1)][(i-1)*81+(j-1)*9+(k-1)] = 1

Aeq11 = np.zeros((81,9*9*9*S))
for i in range (1,10):
    for k in range (1,10):
        for j in range (1,10):
            Aeq11[(i-1)*9+(k-1)][(i-1)*81+(j-1)*9+(k-1)] = 1

Aeq12 = np.zeros((81,9*9*9*S))
for i in range (1,10):
    for j in range (1,10):
        for k in range (1,10):
            Aeq12[(i-1)*9+(j-1)][(i-1)*81+(j-1)*9+(k-1)] = 1

Aeq13 = np.zeros((9,9*9*9*S))
for u in range (1,4):
    for v in range(1,4):
        for i in range (u*3-2,u*3+1):
            for j in range (v*3-2,v*3+1):
                for k in range (1,10):
                    Aeq13[(u-1)*3+(v-1)][(i-1)*81+(j-1)*9+(k-1)] = 1

Aeq = np.vstack((Aeq10,Aeq11,Aeq12,Aeq13))

Beq = np.zeros(81+81+81+9)
for i in range (1,81+81+81+1):
    Beq[i-1] = 1
for i in range (81+81+81+1,81+81+81+9+1):
    Beq[i-1] = 9

bounds = (0,1)

res = linprog(c=c, A_eq=Aeq, b_eq=Beq, bounds=bounds, integrality=np.ones(9*9*9))
x = res['x']
sud = np.zeros((9,9))

for i in range (0,9):
    for j in range (0,9):
        for k in range (0,9):
            if x[i*81+j*9+k] == 1:
                sud [i][j] = k+1

print(sud)
print(x)
`

and everithing looks allright, but in the end im noticing that despite i put bounds to (0,1) and integrality to 1, **my x values can vary above and below bounds**

like that
![image](https://github.com/yalmip/YALMIP/assets/83855659/d001c186-1040-4a0a-b528-84762a768571)

can somebody help me to understand what am i doing wrong?

**Comment by johanlofberg** [comment]
Wrong forum 

(but -10^-16 is 0 to any solver using standard numerics if that what you are asking)
